Sources/Basic/FileSystem.swift

```diff
@@ -351,6 +365,16 @@ private class LocalFileSystem: FileSystem {
             break
         }
     }
+    
+    func writeFileContents(_ path: AbsolutePath, bytes: ByteString, atomically: Bool) throws {
+        // Perform non-atomic writes using the fast path
+        guard atomically else {
+            return try writeFileContents(path, bytes: bytes)
+        }
+        let temp = try TemporaryFile(deleteOnClose: false)
+        try writeFileContents(temp.path, bytes: bytes)
```

Comment:
If write didn't work, need to attempt to delete the temporary file.

Reply:
Fixed